source/develop/release-management/features/storage/cinderlib-integration.html.md

```diff
@@ -83,45 +124,142 @@ The user delete a disk on the storage domain:
   - The Engine calls a Vdsm verb to attach the volume with the data needed. The Vdsm stores locally the attachment details that will be needed for detach, and then return the path.
-  - The Engine uses the returned path for the volume in the VM XML.
+  - The Engine stores the result in the DB and uses the returned path for the volume in the VM XML.
 
 
 #### Stop VM
 
 The user stops a VM with a Cinderlib disk :
-  - The Engine calls a Vdsm verb to detach the volume with the data needed. The Vdsm use the attachment details provided by the Engine and disconnect the volume.
+  - The Engine calls a Vdsm verb to detach the volume with the volume ID. The Vdsm use the ID provided by the Engine, fetches the needed data from its local DB and disconnect the volume.
   - The Engine calls a Cinderlib API to disconnect the volume.
 
 
 ### VDSM
 
-- Add connection information as part of GetCapabilities verb using 'os_brick'.
-- Add an 'attach' verb that will get the details of the volume, attach to the local host, and return the data of the attachment to the Engine.
-- Add a 'detach' verb that will get the details of the details of the attachment and will disconnect the volume
+#### Host.GetCapabilities
+
+The Vdsm uses the Cinder's 'os_brick' library to collect the data. If the 'os_brick' library is not installed on the host, the 'connector_info' will not be available and VMs with MBS disk will not able to run on that host.
+
+Here an example of the 'connector_info'
+
+```json
+"connector_info": {
+        "initiator": "iqn.1994-05.com.acme:b82360563cee",
+        "ip": "10.10.10.1",
+        "platform": "x86_64",
+        "host": "vdsm02",
+        "do_local_attach": false,
+        "os_type": "linux2",
+        "multipath": true
+    }
+```
+
+#### ManagedVolume.attach_volume
+
+When the user wants to run a VM, Engine will first call CinderLib API to expose the volume to the host according to the "connector_info".
+The Storage Management API will provide the connection information needed by the host to attach.  Then Engine calls the ManagedVolume.attach_volume with this information and the volume ID. The structure of the data is according to os_brick/Cinder format.
+
+Here an example of the connection information (iSCSI volume):
+
+```json
+{
+    "driver_volume_type": "iscsi",
+    "data":
+    {
+        "target_lun": 26,
+        "target_iqn": "iqn.2009-01.com.kaminario:storage.k2.22612",
+        "target_portal": "3.2.1.1:3260",
+        "target_discovered": "True"
+    }
+}
+```
+
+Vdsm will call an os_brick API to attach the volume and will return the volume information data to Engine that will persist it in the DB.
+The volume information consists of the volume path that Engine will use in the VM XML and of the volume attachment as returned from os_brick. The structure of the volume attachment data is according to os_brick/Cinder
```

Comment:
already written in line 171

Reply:
This is referring to 2 differents data object. It is important to mention it for both